### PR TITLE
TGP-714 Add accessibility plugin to trader-goods-profile-frontend

### PR DESCRIPTION
**Pull request by besscerule**

### CODE REVIEW

File: a11y/FrontendAccessibilitySpec.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  // Another limitation of the framework is that it can generate Arbitrary[T] but not Arbitrary[T[_]],
+  // so any nested types (like a Play `Form[]`) must similarly be provided by wrapping
+  // a concrete value using `fixed()`.  Usually, you'll have a value you can use somewhere else
+  // in your codebase - either in your production code or another test.
+  // Note - these values are declared as `implicit` to simplify calls to `render()` below
+  // e.g implicit val arbReportProblemPage: Arbitrary[Form[ReportProblemForm]] = fixed(reportProblemForm)
+
+  // This is the package where the page templates are located in your service
+  val viewPackageName = "app/views"
```
Comment (besscerule) [inline a11y/FrontendAccessibilitySpec.scala]: it this correct?

File: a11y/FrontendAccessibilitySpec.scala [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  // This is the layout class or classes which are injected into all full pages in your service.
+  // This might be `HmrcLayout` or some custom class(es) that your service uses as base page templates.
+  val layoutClasses = Seq(classOf[templates.Layout])
```
Comment (besscerule) [inline a11y/FrontendAccessibilitySpec.scala]: is this correct?

**Review by mycp98 — DISMISSED** [review]

**Review by yohan-oc — DISMISSED** [review]

File: a11y/FrontendAccessibilitySpec.scala [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import views.html._
+import views.html.templates.Layout
+import viewmodels.govuk.summarylist._
+import views.html.auth.SignedOutView
+class FrontendAccessibilitySpec extends AutomaticAccessibilitySpec {
+
+  val viewPackageName                                   = "views"
+  val layoutClasses                                     = Seq(classOf[Layout])
+  val appConfig: FrontendAppConfig                      = app.injector.instanceOf[FrontendAppConfig]
+  //implicit val arbConfig: Arbitrary[FrontendAppConfig] = fixed(appConfig)
```
Comment (liandra-giacomine) [inline a11y/FrontendAccessibilitySpec.scala]: remove commented out code

File: a11y/FrontendAccessibilitySpec.scala [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  val list: SummaryList                                 = SummaryListViewModel(rows = Seq.empty)
+  implicit val arbSummaryList: Arbitrary[SummaryList]   = fixed(list)
+  private val booleanForm: Form[Boolean]                = Form("value" -> boolean)
+  implicit val arbForm: Arbitrary[Form[_]]              = fixed(booleanForm)
```
Comment (liandra-giacomine) [inline a11y/FrontendAccessibilitySpec.scala:19]: this is an arbitrary for form of any type but it uses the boolean form, how does this work/do we need this?

Comment (besscerule) [reply a11y/FrontendAccessibilitySpec.scala:19]: It is used for both the boolean and the arbitrary form there are two that need it.

Comment (besscerule) [reply a11y/FrontendAccessibilitySpec.scala:19]: ukims and nirms

**Review by liandra-giacomine — APPROVED** [review]
As discussed, we'll update the form not to be arbitrary once the session service code is merged

**Pull request merged**